Engine/Settings.cs

```diff
@@ -522,6 +522,9 @@ private static object GetSafeValueFromExpressionAst(ExpressionAst exprAst)
                         case "false":
                             return false;
 
+                        case "null":
```

Comment:
Noticed this was missing and quickly added it. Makes sense from a layering perspective -- null should be parse-able but might not be allowed as a configurable value.

Reply:
What about just simplifying the whole code to bool.tryparse ?

Reply:
If one of the possibilities is `null`, then it would probably look like:

```
string varName = varExprAst.VariablePath.UserPath.ToLower();
if (bool.TryParse(varName, out bool val))
{
    return val;
}

if (varName == "null")
{
    return null;
}

throw CreateInvalidDataExceptionFromAst(varExprAst);
```

I'm not sure that's simpler though -- the switch/case has a nice flow to me, and ideally the C# compiler knows better than I do about how to make it work quickly. (Although it looks like [that's not the case](https://stackoverflow.com/a/126507), since I would implement a trie for optimum speed).